t/lib/Test/PG.pm

```diff
@@ -0,0 +1,108 @@
+package Test::PG;
+
+BEGIN {
+    die "PG_ROOT not found in environment.\n" unless $ENV{PG_ROOT};
+    $main::pg_dir = $ENV{PG_ROOT};
+}
+
+use warnings;
+use strict;
+
+use lib "$main::pg_dir/lib"; # only needed if not using prove with -l
+
+
+=head1 Test::PG
+
+This module provides the environment and some generic functions for
+writing tests for PG macros.  Mostly copied from F<t/build_PG_envir.pl>,
+it also redefines Test2's "exists" function from C<E()> to C<EXISTS()>
+to avoid the conflict with WebWork's exponential function.
+Its final action is to load C<PGbasicmacros.pl>.
+
+The reason for the module is that There Is More Than One Way To Do It
+and people develop different styles.  This is my coding style.
+Also we learn the underlying structures better by re-inventing the wheel.
+We just try to make a better wheel, but success is not guaranteed.
+
+This module strives for elegance in reducing boiler plate in test files,
+a minimum of duplicated code and adheres to the principle of least surprise
+by locating modules below the C<t/lib> directory.
+It does not make a decent cup of tea.
+
+=head2 Usage
+
+To test a macro that needs to be loaded, include this preamble
+at the top of your test file and customize.
+
+  use Test2::V0;
+
+  use MyPGLib;   # does your macro need a module?
+
+  use lib 't/lib';
+  use Test::PG;    # setup a minimal WW environment
+
+  loadMacros("parserMyPGMacro.pl");
+
+  Context("Numeric");
+
+And run your test from the $PG_ROOT directory with
+
+  prove -l t/my_macro/my_test.t
+
+
+=head2 TODO
+
+=head3 Quiet warnings in F<t/contexts/trig_degrees.t>
+
+The functions sin, cos and atan2 are redefined and generate warnings.
+C<delete> them from the symbol table before loading the macro.
+
+F<macros/contextTrigDegrees.pl> declares C<$deg> twice, it being
+a required file, the package scope isn't heeded by the second C<my $deg>.
```

Comment:
On this ... all we need to do is delete the second declaration of `$deg`.  The package key word does not provide scope.  The first declaration is visible throughout the file regardless of there begin any later package declarations.